test/unit/workDiv/src/WorkDivForKernelTest.cpp

```diff
@@ -164,97 +161,68 @@ TEMPLATE_LIST_TEST_CASE("getValidWorkDivForKernel.2D", "[workDivKernel]", TestAc
-                          alpaka::TagCpuOmp2Threads,
-                          alpaka::TagFpgaSyclIntel,
-                          alpaka::TagGpuSyclIntel,
-                          alpaka::TagGenericSycl>)
-    {
-        // Get calculated threads per block from the workDiv that was found by examining the kernel function
-        auto const threadsPerBlock = workDiv.m_blockThreadExtent.prod();
-        // Get hard limits
-        auto const threadsPerBlockLimit = props.m_blockThreadCountMax;
-        // Depending on the GPU type or the compiler this test might fail because threadsPerBlock can be less than
-        // threadsPerBlockLimit, which is the max device limit.
-        if(threadsPerBlockLimit == 1)
-            CHECK(threadsPerBlock == static_cast<Idx>(threadsPerBlockLimit));
-        else
-            CHECK(threadsPerBlock < static_cast<Idx>(threadsPerBlockLimit));
-
-        // too many threads per block
-        auto const invalidWorkDiv
-            = WorkDiv{Vec{8, threadsPerGridTestValue / 8}, Vec{20 * threadsPerBlock, 1}, Vec{1, 1}};
-        auto isWorkDivValidForHip = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, invalidWorkDiv);
-        CHECK(isWorkDivValidForHip == false);
-
-        auto const validWorkDiv = WorkDiv{Vec{8, threadsPerGridTestValue / 8}, Vec{1, threadsPerBlock}, Vec{1, 1}};
-        isWorkDivValidForHip = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, validWorkDiv);
-        CHECK(isWorkDivValidForHip == true);
-    }
-    else if constexpr(alpaka::accMatchesTags<
-                          Acc,
-                          alpaka::TagCpuSerial,
-                          alpaka::TagCpuOmp2Blocks,
-                          alpaka::TagCpuTbbBlocks,
-                          alpaka::TagCpuSycl>)
+    // The valid workdiv values for the kernel may change depending on the GPU type and compiler.
+    // Therefore the generated workdiv is not compared to a specific workdiv in this test.
+
+    // Get calculated threads per block from the workDiv that was found by examining the kernel function.
+    Idx const threadsPerBlock = workDiv.m_blockThreadExtent.prod();
+
+    // Get the device limit.
+    Idx const threadsPerBlockLimit = props.m_blockThreadCountMax;
+
+    // Check that the number of threads per block is within the device limit.
+    CHECK(threadsPerBlock <= threadsPerBlockLimit);
+
+    // Check that using the maximum number of threads per block is valid.
+    auto const validWorkDiv
+        = WorkDiv{Vec{8, threadsPerGridTestValue / threadsPerBlock / 8}, Vec{1, threadsPerBlock}, Vec{1, 1}};
+    CHECK(alpaka::isValidWorkDivKernel<Acc>(dev, kernelBundle, validWorkDiv));
+
+    // Check that using too many threads per block is not valid.
+    auto const invalidWorkDiv
+        = WorkDiv{Vec{8, threadsPerGridTestValue / threadsPerBlock / 8}, Vec{2, threadsPerBlock}, Vec{1, 1}};
+    CHECK(not alpaka::isValidWorkDivKernel<Acc>(dev, kernelBundle, invalidWorkDiv));
```

Comment:
My initial observation is that; the check assumes  `2 * threadsPerBlock` will be more than allowed block size. But `threadsPerBlock` is not the max block size, it is a value found by getValidWorkDiv, why  `2 * threadsPerBlock`  should be more than allowed block-size limit for all accelerators?. Anyway I am trying to understand which accelerator is that. Thanks.

Reply:
Seems to be happening at AccCpuOmp2Threads

Reply:
https://github.com/alpaka-group/alpaka/actions/runs/10170404404/job/28129487668?pr=2333

Reply:
My idea was that
  - `threadsPerGridTestValue` is the maximum number of threads per grid (`props.m_blockThreadCountMax * props.m_gridBlockCountMax`)
  - `Vec{8, threadsPerGridTestValue / threadsPerBlock / 8}` × `Vec{2, threadsPerBlock}` should give a total of `2 * threadsPerGridTestValue / threadsPerBlock * threadsPerBlock / 8 * 8` which I expected to be bigger than `threadsPerGridTestValue`

Reply:
That is, I don't expect to exceed the block size, but the total grid size.

Reply:
I checked the old isValidWorkDiv and isValidWorkDivForKernel; what you described is never checked in alpaka it seems.
Both use accDevProps.m_gridBlockExtentMax[i] and accDevProps.m_blockThreadExtentMax[i] values.  They check each extent of extent-array one by one. But there is no  max-threads-per-grid in Device Properties. 

I can add the following check
  `If(gridBlockExtent.prod()*blockThreadExtent.prod() > accDevProps.m_gridBlockCountMax * accDevProps.m_BlockThreadCountMax) return false;
`

Reply:
Ah, I see !

Let me update the PR to test what we actually check in Alpaka first.

Before adding the new check we should check if it actually fails on the various backends, like CUDA.

Reply:
I added it but accDevProps of Omp2 threads is too large.

- Acc: AccCpuOmp2Threads<2,unsigned int>, invalidWorkDiv: {gridBlockExtent: (8, 22369621), blockThreadExtent: (2, 24), threadElemExtent: (1, 1)}
threadsPerBlock:24, threadsPerBlockLimit:64
calculation: if(178956968 *48 > 4294967295 * 64) return false;
check:4294967168 > 4294967232 returns true because accDevProps.m_gridBlockCountMax is too large for Ompthreads;

Reply:
OK, also on the CpuSerial backend, it seems that the maximum grid size is not validated properly.

`props.m_gridBlockCountMax` is `2147483647` (the maximum value for a signed 32-bit integer), but launching a kernel with `4294967294` (twice that) blocks still works:
```
[+] QueueGenericThreadsBlocking
[-] QueueGenericThreadsBlocking
createTaskKernel workDiv: {gridBlockExtent: (4294967294), blockThreadExtent: (1), threadElemExtent: (1)}, kernelFnObj: TestKernelWithManyRegisters
[+] operator()
operator() blockSharedMemDynSizeBytes: 0 B
The sum is 1492.00, the argument is 200
The sum is 1492.00, the argument is 200
The sum is 1492.00, the argument is 200
The sum is 1492.00, the argument is 200
The sum is 1492.00, the argument is 200
The sum is 1492.00, the argument is 200
The sum is 1492.00, the argument is 200
...
```

Reply:
Great info, thanks.